src/lib/device/physical_dev.cpp

```diff
@@ -220,8 +220,10 @@ void PhysicalDev::submit_batch() { m_drive_iface->submit_batch(); }
 
 //////////////////////////// Chunk Creation/Load related methods /////////////////////////////////////////
 void PhysicalDev::format_chunks() {
-    m_chunk_info_slots = std::make_unique< sisl::Bitset >(hs_super_blk::chunk_info_bitmap_size(m_dev_info));
+    m_chunk_info_slots = std::make_unique< sisl::Bitset >(std::max(1u, hs_super_blk::max_chunks_in_pdev(m_dev_info)));
     auto bitmap_mem = m_chunk_info_slots->serialize(m_pdev_info.dev_attr.align_size);
+    HS_REL_ASSERT_LE(bitmap_mem->size(), hs_super_blk::chunk_info_bitmap_size(m_dev_info),
```

Comment:
I see `chunk_info_bitmap_size` round up to 4K, and didnt properly handle the boundary when  `max_chunks_in_pdev % 8 != 0`

however the `bitset->serilize(align_size == 512)`   added up `sizeof(bitset_serialized) ` and align to 512.
```
    static uint64_t chunk_info_bitmap_size(const dev_info& dinfo) {
        // Chunk bitmap area has bitmap of max_chunks rounded off to 4k page
        return sisl::round_up(std::max(1u, hs_super_blk::max_chunks_in_pdev(dinfo) / 8), 4096);
    }
 ```
 ```            const uint64_t num_bits{total_bits()};
            const uint64_t total_words{bitset_serialized::total_words(num_bits)};
            const uint64_t total_bytes{sizeof(bitset_serialized) + sizeof(word_t) * total_words};
            const uint64_t size{(alignment_size > 0) ? round_up(total_bytes, alignment_size) : total_bytes};
 ```

Reply:
so if `hs_super_blk::max_chunks_in_pdev(m_dev_info) == 4096 * 8`, we get 
`hs_super_blk::chunk_info_bitmap_size ==4096`
and 
`m_chunk_info_slots->serialize = round_up (4096 + sizeof(bitset_serialized), 512) = 4608`